sdk/core/src/pipeline_context.rs

```diff
@@ -0,0 +1,51 @@
+use crate::Context;
+
+/// Pipeline internal execution context.
+///
+/// During a pipeline execution, context will be passed from the function
+/// starting the pipeline down to each pipeline policy. This struct it the
+/// one that will actually passed down a pipeline. It's the pipeline
+/// responsibility to specify the BAG generics: it can be anything as
+/// long as it's Send and Sync. For example, Cosmos uses the
+/// PipelineContext to pass the ResourceType down to its
+/// AuthorizationPolicy.
+pub struct PipelineContext<BAG>
+where
+    BAG: Send + Sync,
+{
+    inner_context: Context,
+    bag: BAG,
```

Comment:
Not sure I love the name `bag`. It's really the service specific context.

Reply:
It's a leftover of the "old", property bag based approach. You are right, it's not fitting here. I'll use C everywhere.